searx/engines/__init__.py

```diff
@@ -146,11 +146,29 @@ def update_engine_attributes(engine, engine_data):
             setattr(engine, arg_name, copy.deepcopy(arg_value))
 
 
-def set_language_attributes(engine):
+def set_language_attributes(engine, engine_data):
     # pylint: disable=protected-access
     # assign supported languages from json file
+
     if engine.name in ENGINES_LANGUAGES:
         engine.supported_languages = ENGINES_LANGUAGES[engine.name]
+    elif engine.engine in ENGINES_LANGUAGES:
+        # The key of the dictionary ENGINES_LANGUAGES is the *engine name*
+        # configured in settings.xml.  When multiple engines are configured in
+        # settings.yml to use the same origin engine (python module) these
+        # additional engines can use the languages from the origin engine.
+        # For this use the configured ``engine: ...`` from settings.yml
+        engine.supported_languages = ENGINES_LANGUAGES[engine.engine]
+
+    # For an engine, when there is `language: ...` in the YAML settings, the
+    # engine supports only one language, in this case engine.supported_languages
+    # should contains this value defined in settings.yml
+    language = engine_data.get('language', None)
+    if language:
+        if isinstance(engine.supported_languages, dict):
+            engine.supported_languages = {language : engine.supported_languages[language]}
```

Comment:
This code crashes when `engine.supported_languages[language]` doesn't exist and `engine.supported_languages` is dictionary.

So it crashes with (see [engine_languages.json](https://github.com/searxng/searxng/blob/feb2d4dda575eb1501fb5a8427de364bed7bc179/searx/data/engines_languages.json#L25446-L25451) ) :
```yaml
- engine: google
   language: zh-Hans-CN
   ...
```

But not with (see [engine_languages.json](https://github.com/searxng/searxng/blob/feb2d4dda575eb1501fb5a8427de364bed7bc179/searx/data/engines_languages.json#L237-L240) ):
```yaml
- engine: bing
   language: zh-Hans-CN
   ...
```

I suggest to replace line 168 by:
```python
if isinstance(engine.supported_languages, dict) and language in engine.supported_languages:
```

Reply:
> This code crashes ..

I know, IMO it is a miss configuration when the configured language is not in the supported languages .. I also would like to add an exception when language is not in a list type:
```python
else:
    if language not in engine.supported_languages:
        raise KeyError(language)
    engine.supported_languages = [language] 
```

I looked  for an Exception-Type that fits better, but haven't found one .. may be should use SearxException?

https://github.com/searxng/searxng/blob/adeb084cf4ffc59f1e49b47d049e7541be6dd5dd/searx/exceptions.py#L19-L20

Reply:
`SearxSettingsException` ?

https://github.com/searxng/searxng/blob/master/searx/settings_loader.py#L22-L29